python/ray/data/_internal/datasource/delta_datasink.py

```diff
@@ -0,0 +1,3046 @@
+        account_info["account_name"] = os.getenv("AZURE_STORAGE_ACCOUNT_NAME")
+        account_info["account_key"] = os.getenv("AZURE_STORAGE_ACCOUNT_KEY")
+        account_info["sas_token"] = os.getenv("AZURE_STORAGE_SAS_TOKEN")
+        account_info["tenant_id"] = os.getenv("AZURE_TENANT_ID")
+        account_info["client_id"] = os.getenv("AZURE_CLIENT_ID")
+        account_info["client_secret"] = os.getenv("AZURE_CLIENT_SECRET")
+
+        # Try Ray cluster resources for Anyscale
+        if ray.is_initialized():
+            cluster_resources = ray.cluster_resources()
+
+            # Look for Azure-related resources
+            azure_resources = {
+                k.split(":", 2)[-1]: k.split(":", 1)[1]
+                for k in cluster_resources.keys()
+                if k.startswith("anyscale/azure:")
+            }
+
+            for key, value in azure_resources.items():
+                if key == "storage-account":
+                    account_info["account_name"] = value
+                elif key == "tenant":
+                    account_info["tenant_id"] = value
+                elif key == "subscription":
+                    account_info["subscription_id"] = value
+
+        # Filter out None values
+        return {k: v for k, v in account_info.items() if v is not None}
+
+    @staticmethod
+    def _get_azure_region():
+        """
+        Get Azure region from environment or Ray cluster.
+
+        Returns:
+            Region string or None
+        """
+        import os
+
+        # Try environment variable
+        region = os.getenv("AZURE_REGION") or os.getenv("AZURE_LOCATION")
+        if region:
+            return region
+
+        # Try Ray cluster resources for Anyscale
+        if ray.is_initialized():
+            cluster_resources = ray.cluster_resources()
+            region_resources = [
+                k.split(":")[1]
+                for k in cluster_resources.keys()
+                if k.startswith("anyscale/azure-region:")
+            ]
+            if region_resources:
+                return region_resources[0]
+
+        return None
+
+
+def try_get_deltatable(
+    table_uri: Union[str, Path], storage_options: Optional[Dict[str, str]]
+):
+    """
+    Attempts to get a DeltaTable from the given URI.
+    """
+    from deltalake.table import DeltaTable
+
+    try:
+        return DeltaTable(table_uri, storage_options=storage_options)
+    except Exception as e:
+        return None
+
+
+@dataclass
+class DeltaWriteConfig:
+    """
+    Configuration object for writing data to a DeltaTable.
+    All parameters default to None unless specified.
+    """
+
+    # --- Required for file/block writing ---
+    schema: Optional[pa.Schema] = None
+
+    # --- Partitioning ---
+    partition_cols: Optional[List[str]] = None
+    partition_by: Optional[Union[List[str], str]] = None
+    partition_filters: Optional[List[Tuple[str, str, Any]]] = None
+
+    # --- File writing options ---
```

Comment:
Are these targets, hard limits, or soft limits?

Are these inherited from another code base or how were sizes chosen?

Just noting that data width is not considered here, which obviously will drastically impact file size.